tsdb/executor.go

```diff
@@ -1044,22 +1053,6 @@ func ProcessAggregateDerivative(results [][]interface{}, isNonNegative bool, int
 		}
 	}
 
-	// Check the value's type to ensure it's an numeric, if not, return a 0 result. We only check the first value
```

Comment:
Yeah, that is part of it.  The current code has the side effect of returning early if the first values nil too.